src/engine/mqttControl.js

```diff
@@ -425,80 +427,91 @@ class MqttControl extends EventEmitter{
-                }
-            }
+            const filteredList = this.parseTextFile(splitArgs[1]);
+            const filteredSequence = this.filterSequence(filteredList);
+            const loopAmount = this.setNumberOfLoops(splitForLoopNum[1].trim(), virtualSatIndex, filteredList);   
+            this.broadcastVirtualLightSeq(loopAmount, filteredSequence, virtualSatIndex);
+            this.scheduleResetVirtualSatLights(blackoutSequence, virtualSatIndex);
+            this._timeoutHolder = 0;
+        }
+    }
+
+    static parseTextFile (lightFileName) {
+        const textLightSequence = require(`!raw-loader!../import/SatelliteFiles/${lightFileName}`);
+        let split = '';
+        let seq = '';
+
+        if (typeof textLightSequence === 'string') {
+            split = textLightSequence.split('\n');
+        } else if (typeof textLightSequence === 'object') {
+            const string = textLightSequence.default.toString();
+            split = string.split('\n');
+        }
+        const filtered = split.filter(e => e === 0 || e);
+        seq = filtered.join(',');
+        const stringSplit = seq.split(',');
+        const filteredList = stringSplit.filter(e => e === 0 || e === '' || e);
+        return filteredList;
+    }
+
+    static setNumberOfLoops (loopCount, virtualSatIndex, sequence) {
+        let loops = 0;
+        if (loopCount === '-1') {
+            // infinite loops
+            const data = {
+                satNum: virtualSatIndex,
+                loopState: true,
+                fullSequence: sequence
+            };
+            this.runtime.emit(`SET_VIRTUALSAT_LOOPSTATE`, data);
+            loops = 1;
+        } else {
+            loops = loopCount;
+        }
+        return loops;
+    }
 
-            while (loopAmount > 0) {
-                let positionsCopy = [];
-                const positions = ['#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000'];
-                for (let i = 0; i < totalPos.length; i++) {
-                    positionsCopy = [...positions];
-                    if (totalPos[i].includes('L')) {
-                        if (totalPos[i].includes(',')) {
-                            this.processLCommandList(totalPos[i], positionsCopy, _timeoutIds, virtualSatIndex);
-                        } else {
-                            this.processLCommandSingle(totalPos[i], positionsCopy, _timeoutIds, virtualSatIndex);
-                        }
+    static broadcastVirtualLightSeq (loopAmount, filteredSequence, satNum) {
+        while (loopAmount > 0) {
+            let positionsCopy = [];
+            const positions = ['#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000'];
+            for (let i = 0; i < filteredSequence.length; i++) {
```

Comment:
you know we should probably investigate parsing libs for JS.  nothing to do now, but make a note of it..